From 98254030694ebb070c9af5a4512e5938a9ca1162 Mon Sep 17 00:00:00 2001
From: "tpopela@redhat.com"
 <tpopela@redhat.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Thu, 16 Mar 2017 08:40:29 +0000
Subject: [PATCH] Fix uninitialized public members in Webkit2/UIProcess/API/gtk
 https://bugs.webkit.org/show_bug.cgi?id=169602

Reviewed by Michael Catanzaro.

Found by Coverity scan.

* UIProcess/API/gtk/WebKitMimeInfo.cpp:
* UIProcess/API/gtk/WebKitSettings.cpp:
* UIProcess/API/gtk/WebKitWebViewBase.cpp:

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@214034 268f45cc-cd09-0410-ab3c-d52691b4dbfc

diff --git a/Source/WebKit2/UIProcess/API/gtk/WebKitMimeInfo.cpp b/Source/WebKit2/UIProcess/API/gtk/WebKitMimeInfo.cpp
index af61dc69a8f..87f30ee43d2 100644
--- a/Source/WebKit2/UIProcess/API/gtk/WebKitMimeInfo.cpp
+++ b/Source/WebKit2/UIProcess/API/gtk/WebKitMimeInfo.cpp
@@ -35,7 +35,7 @@ struct _WebKitMimeInfo {
     CString description;
     GRefPtr<GPtrArray> extensions;
 
-    int referenceCount;
+    int referenceCount { 0 };
 };
 
 G_DEFINE_BOXED_TYPE(WebKitMimeInfo, webkit_mime_info, webkit_mime_info_ref, webkit_mime_info_unref)
diff --git a/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp b/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp
index 1b96bf2a0b4..b8617ff4332 100644
--- a/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp
+++ b/Source/WebKit2/UIProcess/API/gtk/WebKitSettings.cpp
@@ -71,8 +71,8 @@ struct _WebKitSettingsPrivate {
     CString pictographFontFamily;
     CString defaultCharset;
     CString userAgent;
-    bool allowModalDialogs;
-    bool zoomTextOnly;
+    bool allowModalDialogs { false };
+    bool zoomTextOnly { false };
 };
 
 /**
diff --git a/Source/WebKit2/UIProcess/API/gtk/WebKitWebViewBase.cpp b/Source/WebKit2/UIProcess/API/gtk/WebKitWebViewBase.cpp
index b057a5a442b..aa878894861 100644
--- a/Source/WebKit2/UIProcess/API/gtk/WebKitWebViewBase.cpp
+++ b/Source/WebKit2/UIProcess/API/gtk/WebKitWebViewBase.cpp
@@ -161,42 +161,42 @@ struct _WebKitWebViewBasePrivate {
     WebKitWebViewChildrenMap children;
     std::unique_ptr<PageClientImpl> pageClient;
     RefPtr<WebPageProxy> pageProxy;
-    bool shouldForwardNextKeyEvent;
-    bool shouldForwardNextWheelEvent;
+    bool shouldForwardNextKeyEvent { false };
+    bool shouldForwardNextWheelEvent { false };
     ClickCounter clickCounter;
     CString tooltipText;
     IntRect tooltipArea;
     GRefPtr<AtkObject> accessible;
-    GtkWidget* authenticationDialog;
-    GtkWidget* inspectorView;
-    AttachmentSide inspectorAttachmentSide;
-    unsigned inspectorViewSize;
+    GtkWidget* authenticationDialog { nullptr };
+    GtkWidget* inspectorView { nullptr };
+    AttachmentSide inspectorAttachmentSide { AttachmentSide::Bottom };
+    unsigned inspectorViewSize { 0 };
     GUniquePtr<GdkEvent> contextMenuEvent;
-    WebContextMenuProxyGtk* activeContextMenuProxy;
+    WebContextMenuProxyGtk* activeContextMenuProxy { nullptr };
     InputMethodFilter inputMethodFilter;
     KeyBindingTranslator keyBindingTranslator;
     TouchEventsMap touchEvents;
     IntSize contentsSize;
 
-    GtkWindow* toplevelOnScreenWindow;
-    unsigned long toplevelFocusInEventID;
-    unsigned long toplevelFocusOutEventID;
-    unsigned long toplevelWindowStateEventID;
-    unsigned long toplevelWindowRealizedID;
+    GtkWindow* toplevelOnScreenWindow { nullptr };
+    unsigned long toplevelFocusInEventID { 0 };
+    unsigned long toplevelFocusOutEventID { 0 };
+    unsigned long toplevelWindowStateEventID { 0 };
+    unsigned long toplevelWindowRealizedID { 0 };
 
     // View State.
-    ActivityState::Flags activityState;
-    ActivityState::Flags activityStateFlagsToUpdate;
+    ActivityState::Flags activityState { 0 };
+    ActivityState::Flags activityStateFlagsToUpdate { 0 };
     RunLoop::Timer<WebKitWebViewBasePrivate> updateActivityStateTimer;
 
-    WebKitWebViewBaseDownloadRequestHandler downloadHandler;
+    WebKitWebViewBaseDownloadRequestHandler downloadHandler { 0 };
 
 #if ENABLE(FULLSCREEN_API)
-    bool fullScreenModeActive;
+    bool fullScreenModeActive { false };
     WebFullScreenClientGtk fullScreenClient;
     GRefPtr<GDBusProxy> screenSaverProxy;
     GRefPtr<GCancellable> screenSaverInhibitCancellable;
-    unsigned screenSaverCookie;
+    unsigned screenSaverCookie { 0 };
 #endif
 
     std::unique_ptr<AcceleratedBackingStore> acceleratedBackingStore;
-- 
2.13.5

